broker/httpd/auth.go

```diff
@@ -0,0 +1,80 @@
+package httpd
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Symantec/cloud-gate/lib/constants"
+)
+
+func randomStringGeneration() (string, error) {
+	const size = 32
+	bytes := make([]byte, size)
+	_, err := rand.Read(bytes)
+	if err != nil {
+		return "", err
+	}
+	return base64.URLEncoding.EncodeToString(bytes), nil
+}
+
+func (s *Server) loginHandler(w http.ResponseWriter, r *http.Request) {
+	userInfo, err := s.authSource.GetRemoteUserInfo(r)
+	if err != nil {
+		s.logger.Println(err)
+		http.Error(w, fmt.Sprint(err), http.StatusInternalServerError)
+		return
+	}
+	if userInfo == nil {
+		s.logger.Println("null userinfo!")
+
+		http.Error(w, "null userinfo", http.StatusInternalServerError)
+		return
+	}
+	randomString, err := randomStringGeneration()
+	if err != nil {
+		s.logger.Println(err)
+		http.Error(w, "cannot generate random string", http.StatusInternalServerError)
+		return
+	}
+
+	expires := time.Now().Add(time.Hour * constants.CookieExpirationHours)
+
+	userCookie := http.Cookie{Name: constants.AuthCookieName, Value: randomString, Path: "/", Expires: expires, HttpOnly: true, Secure: true}
+
+	http.SetCookie(w, &userCookie)
+
+	Cookieinfo := AuthCookie{*userInfo.Username, userCookie.Expires}
+
+	s.cookieMutex.Lock()
+	s.authCookie[userCookie.Value] = Cookieinfo
+	s.cookieMutex.Unlock()
+
+	http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func (s *Server) GetRemoteUserName(w http.ResponseWriter, r *http.Request) (string, error) {
```

Comment:
Please move the public method to api.go and have that call a private method here. This way all public methods are in the api.go file, which makes it easier to see the entire API in a single place.